open-metadata-implementation/adapters/governance-engines-plugins/gaian-ranger-plugin/src/main/java/org/odpi/openmetadata/adapters/governanceenginesplugins/gaianrangerplugin/RangerPolicyResultFilter.java

```diff
@@ -127,155 +282,142 @@ public boolean setQueriedColumns(int[] queriedColumns) {
-            // results or exceptions.
-            // **HACK** for now, skip over columns with NULL data. PURELY for demo support pending a proper
-            // fix
-            authorizer.applyRowFilterAndColumnMasking(queryContext);
-
-            int resultSetColumnIndexOffset = 0;
-            int querySetColumnIndex = 0;
-            int firstValidRow = -1;
-
-            // If there's no rows, masking isn't needed
-            if (null != rows && rows.length > 0) {
-
-                for (int i = 0; i < rows.length; i++) {
-                    if (rows[i] != null) {
-                        firstValidRow = i;
-                        break;
-                    }
-                }
+        if (null != opID && opID.equals(OP_ID_SET_AUTHENTICATED_DERBY_USER_RETURN_IS_QUERY_ALLOWED)) {
 
-                // if all the rows are null return directly
-                if (firstValidRow == -1) return rows;
-
-                while (querySetColumnIndex < queryContext.getColumns().size()
-                        && querySetColumnIndex + resultSetColumnIndexOffset < rows[firstValidRow].length) {
-                    // We ONLY look at the first row - this will fail if the data is null
-                    // instead should be consulting metadata (work needed to resolve)
-                    // BIG HACK warning....
-                    if (rows[firstValidRow][querySetColumnIndex + resultSetColumnIndexOffset].isNull()) {
-                        resultSetColumnIndexOffset++; // increment the fudge factor
-                        continue; // resume with the next expected column
-                    }
+            if (args != null && args.length >= 1 && null != args[0]) {
+                setUserDetailsForQueryContext(args[0]);
+                haveUser = true;
 
-                    if (queryContext.getColumnTransformers().get(querySetColumnIndex) != queryContext.getColumns().get(querySetColumnIndex)) {
-                        // Now do the transformation for each row
-                        for (int j = 0; j < rows.length; j++) {
-                            if (rows[j] == null) continue;
-                            ApplyMasking.redact(rows[j][querySetColumnIndex + resultSetColumnIndexOffset]);
-                        }
+            }
 
-                    }
-                    querySetColumnIndex++;
+            if (!haveUser) {
+                queryContext.setUser(UNKNOWN_USER);
+                logger.logImportant("Unable to retrieve user. Ensure Table functions configured and used in Query. Defaulting to setting user to <UNKNOWN>");
+            }
+        }
+        return null;
+    }
 
-                }
+    private void setUserDetailsForQueryContext(Object arg) {
+        String gaianUser = arg.toString().toLowerCase();
+        queryContext.setUser(gaianUser);
+        logger.logInfo("Found user for query :" + gaianUser);
 
+        if (!gaianUser.equals(DEFAULT_GAIAN_DB_USER)) {
```

Comment:
the standard 'system' gaian db user can be changed - in gaiandb properties, so hardcoding a value is uncomfortable. I presume the intent is to avoid group lookups in some cases, but I think caching and/or asynchronous would be more appropriate.

The other option could be to specify additional configuration via the servicedef (and hence on the ranger server)
such as
 * list of users to avoid ldap lookup
or similar

I think we should record as issue (since mostly people won't change id) or fix.

Reply:
fixed in the last commit